### ISSUE TITLE
_int_malloc: Assertion while training a simple Neural Network / memory corruption

### ISSUE THREAD

**Issue by Shaunakde** [BVLC/caffe#2325]
**tr;dr:** The dataset had an issue with the label - one label was set to '3' which caused the crash. Closing issue. Note to self: Always check the data.
_______________
I am trying to construct a very simple neural network and run some data (H5) through it. 

When I set a batch size of 50, I get the following error:
But the same model runs fine when I use a batch size of 5 (both on train and test)

```
I0415 09:32:37.817755 61149 solver.cpp:315]     Test net output #1: loss = 1.7104 (* 1 = 1.7104 loss)
I0415 09:32:37.823174 61149 solver.cpp:266] Iteration 9800, Testing net (#0)
I0415 09:32:37.832511 61149 solver.cpp:315]     Test net output #0: accuracy = 0.4119
I0415 09:32:37.832576 61149 solver.cpp:315]     Test net output #1: loss = 1.7276 (* 1 = 1.7276 loss)
I0415 09:32:37.838304 61149 solver.cpp:266] Iteration 9900, Testing net (#0)
I0415 09:32:37.847270 61149 solver.cpp:315]     Test net output #0: accuracy = 0.4118
I0415 09:32:37.847393 61149 solver.cpp:315]     Test net output #1: loss = 1.71049 (* 1 = 1.71049 loss)
caffe: malloc.c:3700: _int_malloc: Assertion `victim->fd_nextsize->bk_nextsize == victim' failed.
*** Aborted at 1429115557 (unix time) try "date -d @1429115557" if you are using GNU date ***
PC: @     0x7f5fb32b3cc9 (unknown)
```

And sometimes this error:
```
*** Error in `build/tools/caffe': malloc(): memory corruption: 0x0000000001c85440 ***
*** Aborted at 1429117817 (unix time) try "date -d @1429117817" if you are using
```

My Model is: http://pastebin.com/DHndjL9M
My Solver is: http://pastebin.com/VmqwBXMB

I am running ubuntu 14.04 on a VMplayer VM with 4096BM of RAM allocated 
The data files are: http://we.tl/ZTUPqn61vN

#Update#
If I change the network parameters it works. My network architecture is currently as follows:
*Fails*: Data -> Inner Product (10) -> ReLU -> Inner Product (3) -> Softmax
*Works*: Data -> Inner Product (50) -> ReLU -> Inner Product (9) -> Softmax

#Update2#
If I set the second layer to 3 - No combination works. Tried 3-3, 4-3, 5-3, .... 15-3
But if I set the second layer to anything above 4, it works. so 10-4, 10-5 .... 100-10 all work.

Update: 1000-4 worked too. it just seems to have a problem with 3 outputs in the final layer.

**Comment by Shaunakde** [comment]
So the data turned out to be at fault. Closing this comment. I hope no one has spent too much time solving this. (except me :8ball: )

Some basic notes for all those who search in the future:
* The labels should be zero indexed. So for number of outputs to be valid at 3 - 0,1,2 are the only valid labels.
* The data may contain one off erroneous values. A simple sanity check saves time.

**Comment by roshan-dongre** [comment]
I'm getting the same error- do you have any example code that shows how to fix the problem? Thanks!
